File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // Prompt user for change owed, in dollars
    float dollars = -1;
    while (dollars < 0)
    {
        dollars = get_float("Change oewd: ");
    }

    // Convert the ammount owed to cents (rounded to nearest penny)
    int cents = round(100 * dollars);

    // Innitialize variable "coins"  to count number of coins needed
    int coins = 0;

    // Calculate number of quarters needed, subtracting the ammount paid with quarters from total owed
    int quarters = cents / 25;
    coins += quarters;
    cents -= quarters * 25;

    // Repeat the process for dimes, nickels, and pennys
    int dimes = cents / 10;
    coins += dimes;
    cents -= dimes * 10;

    int nickels = cents / 5;
    coins += nickels;
    cents -= nickels * 5;

    int pennys = cents;
    coins += pennys;
    cents -= pennys;

    // Print number of coins needed
    printf("%i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float

# Ask user for change owed and convert it to cents
dollars = -1
while dollars < 0:
    dollars = get_float("Owed: ")
cents = round(100 * dollars)

# Calculate number of quarters required
# Subtract ammount paid with quarters from total owed
q = int(cents / 25)
cents -= 25 * q

# Repeat the process for dimes, nickels, and pennys
d = int(cents / 10)
cents -= 10 * d

n = int(cents / 5)
cents -= 5 * n

p = int(cents)
cents -= p

# Print number of coins needed
print(int(q + d + n + p))
